File: lab/python/binarios/conversao_binario_recursivo.py
def decimal_to_binarie(numero):
    if numero == 0:
        return ""
    resto = numero % 2
    return str(decimal_to_binarie(numero // 2)) + str(resto)


def decimal_to_binarie2(numero, prefix=True):
    # trata n = 0 só na chamada de topo
    if numero == 0:
        return "0b0" if prefix else "0"

    def rec(x):
        if x == 0:
            return ""               # nada de "0" aqui!
        return rec(x // 2) + str(x % 2)

    bits = rec(abs(numero))
    return ("0b" + bits) if prefix else bits


numero = 13
resultado = decimal_to_binarie(numero)
print(f"{numero} -> {resultado or '0'}")

numero = 10
resultado = decimal_to_binarie2(numero)
print(f"{numero} -> {resultado or '0'}")

File: lab/javascript/binarios/conversao_binario_recursivo.js
function decimal_to_binarie(numero) {
  if (numero === 0) return "";
  const resto = numero % 2;
  return decimal_to_binarie(Math.floor(numero / 2)) + resto.toString();
}

function decimal_to_binarie2(numero, prefix = true) {
  if (numero === 0) return prefix ? "0b0" : "0";

  function rec(x) {
    if (x === 0) return "";
    return rec(Math.floor(x / 2)) + (x % 2).toString();
  }

  const bits = rec(Math.abs(numero));
  return prefix ? "0b" + bits : bits;
}

// Testes
let numero = 13;
let resultado = decimal_to_binarie(numero);
console.log(`${numero} -> ${resultado || "0"}`);

numero = 10;
resultado = decimal_to_binarie2(numero);
console.log(`${numero} -> ${resultado || "0"}`);
